kroxylicious-operator/src/test/java/io/kroxylicious/kubernetes/operator/ProxyReconcilerTest.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator;
+
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+
+import org.assertj.core.api.InstanceOfAssertFactories;
+import org.assertj.core.api.ObjectAssert;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxy;
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxyBuilder;
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxyStatus;
+import io.kroxylicious.kubernetes.api.v1alpha1.kafkaproxystatus.Conditions;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class ProxyReconcilerTest {
+
+    @Mock
+    Context<KafkaProxy> context;
+
+    private AutoCloseable closeable;
+
+    @BeforeEach
+    public void openMocks() {
+        closeable = MockitoAnnotations.openMocks(this);
+    }
+
+    @AfterEach
+    public void releaseMocks() throws Exception {
+        closeable.close();
+    }
+
+    @Test
+    void successfulReconciliationShouldResultInReadyTrueCondition() {
+        // Given
+        long generation = 42L;
+        var primary = new KafkaProxyBuilder()
+                .withNewMetadata()
+                .withGeneration(generation)
+                .withName("my-proxy")
+                .endMetadata()
+                .build();
+
+        // When
+        var updateControl = new ProxyReconciler().reconcile(primary, context);
+
+        // Then
+        assertThat(updateControl.isPatchStatus()).isTrue();
+        var statusAssert = assertThat(updateControl.getResource()).isNotNull()
+                .extracting(KafkaProxy::getStatus);
+        statusAssert.extracting(KafkaProxyStatus::getObservedGeneration).isEqualTo(generation);
+        ObjectAssert<Conditions> first = statusAssert.extracting(KafkaProxyStatus::getConditions, InstanceOfAssertFactories.list(Conditions.class))
+                .first();
+        first.extracting(Conditions::getObservedGeneration).isEqualTo(generation);
+        first.extracting(Conditions::getLastTransitionTime).isNotNull();
+        first.extracting(Conditions::getType).isEqualTo("Ready");
+        first.extracting(Conditions::getStatus).isEqualTo(Conditions.Status.TRUE);
+        first.extracting(Conditions::getMessage).isEqualTo("");
+        first.extracting(Conditions::getReason).isEqualTo("");
```

Comment:
The other option to this collection of `extracting` is something like 
```
var statusAssert = assertThat(updateControl.getResource())
.isNotNull()
.satisfies(kp -> assertThat(kp.getStatus()).SOMETHING_CLEVER()
```

Not sure if it would help here or not, but I only realised it was an option recently and it helped simplify some tests I was writing.

Reply:
Thanks, I'll try to remember that. None of these APIs feels _nice_, and it's not really clear to me whether one is better than the other.

Reply:
`satisfies` might help more with the assertion on the condition(s) rather than the status, but yeah none of them are great.